### PR TITLE
docs/command-line-interface: Add Runtime CLI Spec

### PR DESCRIPTION
**Pull request by wking**
Transferred from opencontainers/runtime-spec#513 [as discussed in yesterday's meeting][1].  @mrunalp gave me some pointers on how he wanted this landed [here][2] and [here][3].

I'm not squashing this down into a single commit, because it's a lot easier for me to manage revisions if I keep a branch in the [stand-alone repository][4].  I would encourage reviewers (CC @RobDolinMS and @stevvooe, who expressed an interest in reviewing this stuff) to file issues/PRs against that repository so this PR doesn't get bogged down into the same morass as the runtime-spec PR; this PR would get updated to pull those in as I land the sub-PRs in the stand-alone repo.  @mrunalp thinks we might end up [squashing this just before merging][5], and how that goes is clearly up to the runtime-tools maintainers, but squashing (if it happens at all) should be after substantive review settles.

The WIP tag in the summary is because I'm still not sure where runtime-tools maintainers want [the JSON Schema][6] from the runtime-spec PR.

@mrunalp was also [open to some compliance language][7], but I haven't figured that out yet because the pattern [set in RFC 2616][8] (which runtime-spec followed) wasn't even repeated by [RFC 7230][9] (which obsoleted RFC 2616).  I need some time to compare the differences in the RFC wording before I can float appropriate wording for the command line API.

[1]: http://ircbot.wl.linuxfoundation.org/meetings/opencontainers/2017/opencontainers.2017-02-08-22.00.log.html#l-125
[2]: http://ircbot.wl.linuxfoundation.org/eavesdrop/%23opencontainers/%23opencontainers.2017-02-09.log.html#t2017-02-09T22:21:59
[3]: http://ircbot.wl.linuxfoundation.org/eavesdrop/%23opencontainers/%23opencontainers.2017-02-10.log.html#t2017-02-10T00:02:44
[4]: https://github.com/wking/oci-command-line-api
[5]: http://ircbot.wl.linuxfoundation.org/eavesdrop/%23opencontainers/%23opencontainers.2017-02-10.log.html#t2017-02-10T00:18:53
[6]: https://github.com/opencontainers/runtime-spec/pull/513/files#diff-751d571a4cba0b0155c3d3bf793e08e8R13
[7]: http://ircbot.wl.linuxfoundation.org/eavesdrop/%23opencontainers/%23opencontainers.2017-02-10.log.html#t2017-02-10T00:18:25
[8]: https://tools.ietf.org/html/rfc2616#section-1.2
[9]: https://tools.ietf.org/html/rfc7230#section-2.5

### CODE REVIEW

**Comment by wking** [comment]
Some CI errors from the merges.  I'll see what I can do about those later tonight.

**Comment by wking** [comment]
CI tests fixed with 8e5c51a → b5deac1 (no change in the tree content, just the merge commit messages).

File: docs/command-line-interface.md [hunk 1]
```diff
@@ -0,0 +1,266 @@
+# OCI Runtime Command Line Interface
+
+This file defines the OCI Runtime Command Line Interface version 1.0.0.
```
Comment (crosbymichael) [inline docs/command-line-interface.md]: Could you maybe add a sentence stating that this interface is specified as a test harness for compliance testing and runtimes wishing to submit for OCI runtime compliance must implement this interface for testing to be possible

Comment (wking) [inline docs/command-line-interface.md]: > Could you maybe add a sentence stating that this interface is specified as a test harness for compliance testing…

Done with b5deac1 → 4771073, borrowing from the very [similar][1] [approach][2] I'd taken in opencontainers/runtime-spec#513.

[1]: https://github.com/opencontainers/runtime-spec/pull/513/files#diff-0c1e5a5baa5649945697a530e09acbf5R4
[2]: https://github.com/opencontainers/runtime-spec/pull/513/files#diff-958e7270f96f5407d7d980f500805b1bR12

**Comment by wking** [comment]
I've taken a stab at compliance language with afdf059 → 6d8cfe2 (with the compliance language coming in with be7de4f).

**Comment by crosbymichael** [comment]
LGTM

**Comment by mrunalp** [comment]
@wking Can you squash the commits before merge?

**Comment by wking** [comment]
On Wed, Mar 15, 2017 at 04:08:28PM -0700, Mrunal Patel wrote:
> @wking Can you squash the commits before merge?

I'd still rather have the maintainers do the squash if they want
things squashed (GitHub should give you a button for it [1], and you
can do it from the command line with ‘git merge --squash …’), because
it makes ‘git blame …’ pretty useless [2].

I've just pushed 6d8cfe2 → 7eaa0b8 adding the JSON Schema and removing
the WIP prefix.

I'm not aware or (or maybe just don't remember ;) any other issues
with the current content floated in 7eaa0b8 outside of the “the Unix
socket bits will probably not work on non-POSIX systems”.  Ping
@RobDolinMS again on that score.

[1]: https://github.com/blog/2141-squash-your-commits
[2]: http://ircbot.wl.linuxfoundation.org/eavesdrop/%23opencontainers/%23opencontainers.2017-02-10.log.html#t2017-02-10T00:23:41

**Comment by wking** [comment]
Pushed 7eaa0b8 → 841fcac fixing a broken link into `schema/README.md`.

File: schema/socket-terminal-request.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
+{
+  "description": "Open Container Runtime Socket Teminal Request Schema",
```
Comment (mrunalp) [inline schema/socket-terminal-request.json]: typo: Teminal -> Terminal

Comment (wking) [inline schema/socket-terminal-request.json]: Thanks :).  Fixed with 841fcac → ebae480.

**Comment by q384566678** [comment]
LGTM
ping @mrunalp @crosbymichael @liangchenye @Mashimiao

**Comment by liangchenye** [comment]
LGTM  and I'll squash the commits while merging it.

**Pull request merged**